unittest/client/src/ilm_common_unittests.cpp

```diff
@@ -0,0 +1,374 @@
+ */
+TEST_F(IlmCommonTest, ilm_init_cannotGetDisplay)
+{
+    mpp_wlDisplays[0] = nullptr;
+    SET_RETURN_SEQ(wl_display_connect, mpp_wlDisplays, MAX_NUMBER);
+    ASSERT_EQ(ILM_FAILED, ilm_init());
+    ASSERT_EQ(wl_display_connect_fake.call_count, 1);
+    ASSERT_EQ(wl_display_connect_fake.return_val_history[0], nullptr);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_init_getFailedOnilmControl_init
+ * @brief               A failure of ilm_init() when ilmControl_init() returns ILM_FAILED
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_display_connect() doesn't return null object
+ *                      -# Mocking the ilmControl_init() returns ILM_FAILED
+ *                      -# Calling the ilm_init()
+ *                      -# Verification point:
+ *                         +# ilm_init() must return ILM_FAILED
+ *                         +# ilmControl_init() must be called once time and return ILM_FAILED
+ */
+TEST_F(IlmCommonTest, ilm_init_getFailedOnilmControl_init)
+{
+    mpp_wlDisplays[0] = (struct wl_display*)&m_wlDisplayFakePointer;
+    SET_RETURN_SEQ(wl_display_connect, mpp_wlDisplays, MAX_NUMBER);
+    mp_ilmErrorType[0] = ILM_FAILED;
+    SET_RETURN_SEQ(ilmControl_init, mp_ilmErrorType, MAX_NUMBER);
+    ASSERT_EQ(ILM_FAILED, ilm_init());
+    ASSERT_EQ(ilmControl_init_fake.call_count, 1);
+    ASSERT_EQ(mp_ilmErrorType[0], ilmControl_init_fake.return_val_history[0]);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_init_getSuccess
+ * @brief               A success of ilm_init() when wl_display_connect() doesn't return null object
+ *                      and ilmControl_init() returns ILM_SUCCESS
+ * @test_procedure Steps:
+ *                      -# Call mock_ilmInitSuccess() to mocking the wl_display_connect()
+ *                         doesn't return null object and mocking the ilmControl_init() returns ILM_SUCCESS
+ *                      -# Calling the ilm_init()
+ *                      -# Verification point:
+ *                         +# ilm_init() must return ILM_SUCCESS
+ *                         +# wl_display_connect() must be called once time
+ *                         +# ilmControl_init() must be called once time and retrun ILM_SUCCESS
+ */
+TEST_F(IlmCommonTest, ilm_init_getSuccess)
+{
+    mock_ilmInitSuccess();
+    ASSERT_EQ(ILM_SUCCESS, ilm_init());
+    ASSERT_EQ(wl_display_connect_fake.call_count, 1);
+    ASSERT_EQ(ilmControl_init_fake.call_count, 1);
+    ASSERT_EQ(ILM_SUCCESS, ilmControl_init_fake.return_val_history[0]);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_init_manyTimes
+ * @brief               The number of calls to ilm_init() and ilm_destroy() must correspond.
```

Comment:
rephrase to something like: Test case to ensure the number of calls to ilm_init() ...